syntax = "proto3";

import "google/protobuf/timestamp.proto";

package postservice;

service PostService {
  rpc CreatePost(CreatePostRequest) returns (PostResponse);
  rpc GetPost(GetPostRequest) returns (PostResponse);
  rpc UpdatePost(UpdatePostRequest) returns (PostResponse);
  rpc DeletePost(DeletePostRequest) returns (DeletePostResponse);
  rpc ListPosts(ListPostsRequest) returns (ListPostsResponse);
}

message CreatePostRequest {
  string title = 1;
  string content = 2;
  string creator_user_id = 3;
  bool is_private = 4;
  repeated string tags = 5;
}

message GetPostRequest {
  string post_id = 1;
}

message UpdatePostRequest {
  string post_id = 1;
  string title = 2;
  string content = 3;
  bool is_private = 4;
  repeated string tags = 5;
}

message DeletePostRequest {
  string post_id = 1;
  string user_id = 2;
}

message DeletePostResponse {
  bool success = 1;
}

message ListPostsRequest {
  int32 page = 1;
  int32 per_page = 2;
}

message ListPostsResponse {
  repeated PostResponse posts = 1;
  int32 total = 2;
}

message PostResponse {
  string post_id = 1;
  string title = 2;
  string content = 3;
  string creator_user_id = 4;
  google.protobuf.Timestamp created_at = 5;
  google.protobuf.Timestamp updated_at = 6;
  bool is_private = 7;
  repeated string tags = 8;
}
